hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/file/FileDeleteSuite.java

```diff
@@ -50,27 +49,21 @@ public static void main(String... args) {
 
     @Override
     public List<HapiSpec> getSpecsInSuite() {
-        return allOf(positiveTests(), negativeTests());
-    }
-
-    private List<HapiSpec> positiveTests() {
-        return Arrays.asList(getDeletedFileInfo());
-    }
-
-    private List<HapiSpec> negativeTests() {
-        return Arrays.asList(canDeleteWithAnyOneOfTopLevelKeyList());
+        return List.of(getDeletedFileInfo(), canDeleteWithAnyOneOfTopLevelKeyList());
     }
 
+    @HapiTest
     private HapiSpec canDeleteWithAnyOneOfTopLevelKeyList() {
         KeyShape shape = listOf(SIMPLE, threshOf(1, 2), listOf(2));
         SigControl deleteSigs = shape.signedWith(sigs(ON, sigs(OFF, OFF), sigs(ON, OFF)));
 
-        return defaultFailingHapiSpec("CanDeleteWithAnyOneOfTopLevelKeyList")
+        return defaultHapiSpec("CanDeleteWithAnyOneOfTopLevelKeyList")
```

Comment:
Why is the name changed here? Did you run the mono code test to make sure this is correct with this change?

Reply:
because it was part of the negative group and we moved it to be part of the positive otherwise it passed but Intelliji show it as failed.

Reply:
Did you try to run, application.run (old mono code way to test), then run FileDeletSuite.main() to see if the test still pass?

Reply:
I thought we were not supposed to change the test code itself besides adding @HapiTest after each test was passed. Yes, this one is using defaultFailingHapiSpec() instead of defaultHapiSpec(). There are other test suite also use this. We probably should add this handling in HapiTestEngine instead of changing the suite test code.

Reply:
It's not defaultFailingHapiSpec that's the problem here.  It's the combination of using defaultFailingHapiSpec and also placing the test in `negativeTests`.  That combination creates a situation where the test reports FAILURE, but succeeds, and reports SUCCESS, but fails.

We cannot make HapiTestEngine do that without some really ugly changes, and it's a bad setup regardless as it always reports the opposite of what is true (which creates a high probability someone eventually "fixes" it by making the code do the opposite of intended).

Yes, we should not change tests, in general, but this is an exception case where the test was setup badly to begin with, and all we do here is fix that setup to a sane structure so that both HapiTest and MonoService either succeed together or fail together.

Reply:
negativeTests() doesn't do anything except putting the test in the Arrays and return a list, and probably telling you the test expects "expected.final.status" will be "FAILED" as defaultFailingHapiSpec() uses. Default "expected.final.status" is "PASSED" in which defaultHapiSpec() uses. When you run this in mono code, you don't see FAILURE, but succeeds and only see this in HapiTestEngine run. If you change code this way, there are other two suite tests will have to do the same.

Reply:
We observed different behavior, the test was consistently reporting failure, regardless of how it was run, but the suite succeeded.  The cleaned up lists above fixed that behavior so it failed if it reported FAILURE, and succeeded if it reported SUCCESS.  Then this line changed to make it actually succeed when it was supposed to succeed and fail when it was supposed to fail.
Not sure what dark ritual in the HapiSpecSuite was causing all that, but this set of changes fixed it.
The other suites only have to change if they are failing incorrectly.  This change was necessary, not simply desired.

Reply:
Here is the prior behavior:
```
2023-08-24 12:42:51.145 INFO   282  HapiTxnOp - 'CanDeleteWithAnyOneOfTopLevelKeyList' -  HapiFileDelete{sigs=3, node=0.0.3, fileName=test} Wrong actual status INVALID_SIGNATURE, expected SUCCESS
2023-08-24 12:42:51.145 WARN   262  HapiSpecOperation - 'CanDeleteWithAnyOneOfTopLevelKeyList' - HapiFileDelete{sigs=3, node=0.0.3, fileName=test} failed - Wrong status! Expected SUCCESS, was INVALID_SIGNATURE!
2023-08-24 12:42:51.482 INFO   481  HapiSpec - 'CanDeleteWithAnyOneOfTopLevelKeyList' - final status: FAILED!
```
Now perhaps this was just broken in several levels (and it might not even run in current E2E or integration tests).  Regardless the change Lev made here seems to make the test correct and run correctly in both Modularized and Mono Services.

Reply:
This is the output from HapiTestEngine run.

Reply:
The PASSED_UNEXPECTEDLY is what was causing issues, yes.
We couldn't figure out what made that translation from "this should have failed, but did not fail" to "it's fine".
Interesting, the Mono code returns a SUCCESS value if you debug it, rather than INVALID_SIGNATURE, so something was causing `defaultFailingHapiSpec` to accept a success on the test that *should have failed* and translating to PASSED_UNEXPECTEDLY, which is kind of "we should have failed, but didn't, so we'll say the test passed" and seems completely broken.
If we matched the modular code to mono, the HapiTest failed, if we returned the wrong thing, the HapiTest succeeded, and that was clearly not correct.
Changing the spec call to `defaultHapiSpec` makes BOTH versions return PASSED (neither returns PASSED_UNEXPECTEDLY), and that seems more correct than one passing through a backdoor that causes the other to accept incorrect code.

If we have other PASSED_UNEXPECTEDLY items, we definitely should investigate to see if something is broken in the test definition (as was the case here) or if we have a possible bug that the testing doesn't report (because PASSED_UNEXPECTEDLY is not reported as a failure).

Reply:
Short summary:  This test was unequivocally broken and regardless of HapiTest or not, needed to be fixed.  It is fixed now.
While we do not want to change the test suites unless absolutely necessary, it is necessary in this case.

Reply:
I think we should check with @tinker-michaelj or @Neeharika-Sompalli about this defaultFailingHapiSpec().

Reply:
@tinker-michaelj can you comment on it?

Reply:
@povolev15 the new test looks correct to me. 💯

Reply:
@iwsimon please let me know if I can resolve it

Reply:
LGTM